### DISCUSSION TITLE
Secp256k1 library in pure assembler(very fast implementation of arithmetic on curve secp256k1 on x86_64)

### DISCUSSION THREAD

**Discussion by alekssolov** [community/community#114464] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I would like to speeding up work secp256k1 in KeyHunt from albertobsd (https://github.com/albertobsd/keyhunt)

I found this: https://github.com/piggypiggy/secp256k1-x64

This library aims to provide the most efficient implementation of secp256k1 curve arithmetic.

I can compile it, but how convert Int to Point?

Openssl contains similar functions: https://www.openssl.org/docs/man1.0.2/man3/BN_mod_mul_montgomery.html

For example, function secp256k1_sqr_mont (the fastest, according to the developer) i would like to use in SECP256K1.cpp:

Point Secp256K1::ComputePublicKey(Int *privKey) {
Point Q;
long long unsigned int x[4]; //result of the montgomery square
Int result;
Q.Clear();
secp256k1_sqr_mont(x, (const long long unsigned int*)(&(privKey->bits64)));
result.Set32Bytes((unsigned char*)x); // Set the result
??????????? result to Q ???????????
Q.Reduce();
return Q;
}

**Comment by queenofcorgis** [comment]
If you're asking how to use or improve the piece of software that you found in a repository on GitHub, the best way to contact the maintainers of that software is to:

- Check the README for instructions on how to operate the software and pointers to documentation or troubleshooting info.
- Check the [SUPPORT file](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-support-resources-to-your-project), if one exists, for instructions on how to best contact the maintainers for support.
- Check the [CONTRIBUTING guide](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors), if one exists. Sometimes if there isn't a SUPPORT file, the CONTRIBUTING guide will give instructions on how to contact the maintainers for support.

All of these documents can be found, if they exist, in the repository where you found the software itself.

**Reply by alekssolov** [reply]
not what i need
